### PR TITLE
add Instrumental.disconnect() and optional socket timeout

### PR DESCRIPTION
**Pull request by pmidge**
This is the initial cut of my changes and what I'm testing with.

### CODE REVIEW

File: lib/instrumental.js [hunk 1]
```diff
@@ -15,6 +15,7 @@ class Instrumental {
     this.apiKey  = options.apiKey;
     this.enabled = options.enabled === undefined ? true : options.enabled;
     this.host    = options.host || 'collector.instrumentalapp.com';
+    this.timeout = (options.timeout === undefined) ? 0 : (options.timeout || 5 * 60 * 1000);
```
Comment (janxious) [inline lib/instrumental.js]: Might consider pulling this out into a `const` (and probably that `host` default should be too)

Comment (pmidge) [inline lib/instrumental.js]: done

**Comment by pmidge** [comment]
The default timeout assignment was unnecessary. The behavior I wanted essentially reduces down to _falsy_ means "no timeout" and any other value is assumed to be a valid millisecond timeout. In other words, this is enough: `this.timeout = options.timeout;`

So:

`options.timeout = (0 | undefined | null)` causes no timeout to be set.
`options.timeout = [some int]` causes a timeout to be set.

I'll do a final update to the pull request once my testing's done.

**Comment by pmidge** [comment]
testing on my end looks good.

**Comment by janxious** [comment]
Originally the intent was to have a default timeout, too. It seems like that would be:

```javascript
this.timeout = options.timeout || DEFAULT_TIMEOUT
```

Curious why the removal. Otherwise this looks 👍 

----- 

Fixes #16

**Comment by pmidge** [comment]
@janxious sorry, see my comment before the commit for rationale. TL;DR: unneeded IMHO.

**Comment by janxious** [comment]
New version has been cut as 2.1.0. Thanks for the work on this @pmidge 💃

**Comment by pmidge** [comment]
No problem! I've picked up the new version into my environment.

**Pull request merged**